model/clusters_mgmt/v1/machine_pool_type.model

```diff
@@ -33,6 +33,9 @@ class MachinePool {
 	// The subnets upon which the nodes are created.
 	Subnets []String
 
+	// Optional keys and values that the machine pool provisioner will add as tags to all AWS resources it creates.
```

Comment:
I think this rarely have the restrictions on the api-model directly, is there some example to understand how the open api spec will be generated?

Reply:
If we cannot put the restrictions in the api-model / openapi directly, which I think we cannot due to the limitations of the metamodel, we should at least document it for the consumers of the API which it was why I was proposing documenting it.

Aside from that, on CS side we will probably want to add at least some of those validations so it fails earlier in the lifecycle

Reply:
@miguelsorianod I've changed the description to "AWS tags" for more clarity. On the CS side, we already have added validations for these tags: https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/7710

Reply:
Perfect thanks.

Even with the validations implemented I think there is value in documenting the restrictions so the consumers of the API are aware of them

Reply:
that sounds good to me. where would be the best place to document the restrictions? do you mean in this repo or somewhere else?

Reply:
In this repository yes.

That's because then the OpenAPI document is generated from the contents defined here.

Documenting it in the API types of the CS repository itself is also a good idea.

Reply:
@miguelsorianod done! also added documentation for validations on the node pool tags